util/check_tool_requirements.py

```diff
@@ -223,6 +229,21 @@ def to_semver(self, version, from_req):
         return '.'.join(m.group(1, 2, 3))
 
 
+class VivadoToolReq(ToolReq):
+    tool_cmd = ['vivado', '-version']
+    version_regex = re.compile(r'Vivado v(.*)\s')
+
+    def to_semver(self, version, from_req):
+        # Regular Vivado releases just have a major and minor version.
```

Comment:
Shouldn't this comment go next to the bit that uses `m.group(3) or '0'`?

Reply:
Fair point. I added a comment explaining the optional patch number in the regex and then moved the comment about setting it to zero to where you suggested.